doc/hardware/porting/board_porting.rst

```diff
@@ -133,57 +171,127 @@ board directory to a separate repository once it's working.)
-  similar schemes are common for upstream vendor boards.
 
 Your board directory should look like this:
 
 .. code-block:: none
 
    boards/<ARCH>/plank
+   ├── board.yml
    ├── board.cmake
    ├── CMakeLists.txt
    ├── doc
    │   ├── plank.png
    │   └── index.rst
-   ├── Kconfig.board
+   ├── Kconfig.plank
    ├── Kconfig.defconfig
    ├── plank_defconfig
+   ├── plank_<identifier>_defconfig
    ├── plank.dts
+   ├── plank_<identifier>.dts
+   ├── plank_<identifier>.overlay
    └── plank.yaml
 
 Replace ``plank`` with your board's name, of course.
 
 The mandatory files are:
 
-#. :file:`plank.dts`: a hardware description in :ref:`devicetree
-   <dt-guide>` format. This declares your SoC, connectors, and any
-   other hardware components such as LEDs, buttons, sensors, or communication
-   peripherals (USB, BLE controller, etc).
+#. :file:`board.yml`: a YAML file describe the highlevel meta data of the board
+   such as the board name, SoCs, and variants.
+   Cpusets for multi-core SoCs are not described in this file as they are
+   inherited from the SoC's YAML description.
+
+#. :file:`plank.dts` or :file:`plank_<identifier>.dts`: a hardware description
+   in :ref:`devicetree <dt-guide>` format. This declares your SoC, connectors,
+   and any other hardware components such as LEDs, buttons, sensors, or
+   communication peripherals (USB, BLE controller, etc).
+
+#. :file:`Kconfig.plank`: the base software configuration for selecting SoC and
+   other board and SoC related setting. No Kconfig settings outside the board
+   and SoC tree must be selected. To select general Zephyr Kconfig settings the
+   :file:`Kconfig` file shall be used.
 
-#. :file:`Kconfig.board`, :file:`Kconfig.defconfig`, :file:`plank_defconfig`:
-   software configuration in :ref:`kconfig` formats. This provides default
-   settings for software features and peripheral drivers.
 
 The optional files are:
 
+- :file:`plank_<identifier>.overlay`: a hardware description overlay in
+  :ref:`devicetree <dt-guide>` format. This file is used as overlay to a
+  :file:`plank.dts` baseline and allow you to adjust the base devicetree file
+  with adjustments according to the SoC or board variant.
+- :file:`Kconfig`, :file:`Kconfig.defconfig` software configuration in
+  :ref:`kconfig` formats. This provides default settings for software features
+  and peripheral drivers.
+- :file:`plank_defconfig` and :file:`plank_<identifier>_defconfig`: software
+  configuration in Kconfig ``.conf`` format.
 - :file:`board.cmake`: used for :ref:`flash-and-debug-support`
 - :file:`CMakeLists.txt`: if you need to add additional source files to
-  your build.
+  `your build.
 - :file:`doc/index.rst`, :file:`doc/plank.png`: documentation for and a picture
   of your board. You only need this if you're :ref:`contributing-your-board` to
   Zephyr.
 - :file:`plank.yaml`: a YAML file with miscellaneous metadata used by the
```

Comment:
we already today have a twister yaml for each variant. It's just that each variant is implemented as a board.

So nothing changes short term in this regard, but long term it will be possible to cleanup the twister yaml and then unify the n-yaml files, but this work is outside scope of current porting effort.